File: codewars/js/convert_string_to_camel_case.js
// https://www.codewars.com/kata/517abf86da9663f1d2000003/train/javascript

function toCamelCase(str) {
    let arr = [];
    let temp = "";

    for (let i = 0; i < str.length; ++i) {
        if (!["_", "-"].includes(str[i])) {
            temp += str[i];
        } else {
            arr.push(temp);
            temp = "";
        }
    }
    arr.push(temp);

    let camelCase = arr[0];

    for (let i = 1; i < arr.length; ++i) {
        for (let j = 0; j < arr[i].length; ++j) {
            if (j == 0) {
                camelCase += arr[i][j].toUpperCase();
            } else {
                camelCase += arr[i][j];
            }
        }
    }

    return camelCase;
}

console.log(toCamelCase("the-stealth-warrior"));

File: codewars/java/ConvertStringToCamelCase.java
// https://www.codewars.com/kata/517abf86da9663f1d2000003/train/java

public class ConvertStringToCamelCase {
	public static void main(String[] args) {
		String s = "the_Stealth_Warrior";
		s = "The_Stealth-Warrior";
		System.out.println(toCamelCase(s));

	}

	static String toCamelCase(String s) {
		String[] ssplit = s.replaceAll("[-]", "_").split("_");

		String camelCase = new String();
		for (int i = 0; i < ssplit.length; i++) {
			if (i == 0) {
				camelCase += ssplit[0];
			}

			if (i != 0) {
				camelCase += Character.toString(ssplit[i].charAt(0)).toUpperCase();
				camelCase += ssplit[i].substring(1, ssplit[i].length());
			}
		}

		return camelCase;
	}
}
